gadgets/trace_open.bpf.c

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2019 Facebook
+// Copyright (c) 2020 Netflix
+#include <vmlinux/vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include "trace_open.h"
+
+#define TASK_RUNNING	0
+
+const volatile __u64 min_us = 0;
+const volatile pid_t targ_pid = 0;
+const volatile pid_t targ_tgid = 0;
+const volatile uid_t targ_uid = INVALID_UID;
+const volatile bool targ_failed = false;
+
+// we need this to make sure the compiler doesn't remove our struct
+const struct event *unusedevent __attribute__((unused));
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, 10240);
+	__type(key, u32);
+	__type(value, struct args_t);
+} start SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+	__uint(key_size, sizeof(u32));
+	__type(value, struct event);
+	//__uint(value_size, sizeof(u32));
+} print_events SEC(".maps");
+
+static __always_inline bool valid_uid(uid_t uid) {
+	return uid != INVALID_UID;
+}
+
+static __always_inline
+bool trace_allowed(u32 tgid, u32 pid)
+{
+	u64 mntns_id;
```

Comment:
```suggestion
	mnt_ns_id_t mntns_id;
```

(and other in gadgets)

Reply:
I dropped the commit introducing that change and used a different approach to avoid creating too much noise.